app/src/main/java/swati4star/createpdf/adapter/MergeFilesAdapter.java

```diff
@@ -46,10 +50,102 @@ public ViewMergeFilesHolder onCreateViewHolder(@NonNull ViewGroup parent, int vi
 
     @Override
     public void onBindViewHolder(@NonNull ViewMergeFilesHolder holder, int position) {
-        boolean isEncrypted = mPDFUtils.isPDFEncrypted(mFilePaths.get(position));
-        holder.mFileName.setText(FileUtils.getFileName(mFilePaths.get(position)));
-        holder.mEncryptionImage.setVisibility(isEncrypted ? View.VISIBLE : View.INVISIBLE);
+        boolean isPdfFile = isPdfFile(mFilePaths.get(position));
+        if (isPdfFile) {
+            boolean isEncrypted = mPDFUtils.isPDFEncrypted(mFilePaths.get(position));
+            holder.mFileName.setText(FileUtils.getFileName(mFilePaths.get(position)));
+            holder.mEncryptionImage.setVisibility(isEncrypted ? View.VISIBLE : View.INVISIBLE);
+        }
+    }
+
+    /**
+     * check whether the file is a real pdf file or not (whether the file has the header of pdf file)
+     */
+    public static boolean isPdfFile(String filePath) {
+        String header = getFileHeader(filePath);
```

Comment:
#1056 also contains this function as a fix so what I want to say is that #1062  = commits of #1056 (`getFileHeader() method) + `getLastLine()` method.

So because #1062 already contains what present in #1056 so just close #1056 and #1062 will solve all those three issues.

Hope you understood.

Reply:
Ok, I will deal with it.